src/external/clinical/index.ts

```diff
@@ -26,15 +26,32 @@ import _ from 'lodash';
 import { getEgoToken } from '../../external/ego';
 import Logger from '../../logger';
 import { ClinicalDonor } from './types';
+import AsyncCache from '../../utils/asyncCache';
 const logger = Logger('Clinical');
 
+// Cache results of fetching donor data from clinical service. Keep cached results for 5 minutes.
+const DONOR_FETCH_CACHE = AsyncCache(
+  (inputs: { programId: string; donorId: string }) => fetchDonor(inputs.programId, inputs.donorId),
+  { expiryTime: 5 * 60 * 1000 },
+);
+
+/**
+ * Retrieve donor information from clinical service with cached results to prevent redundant fetching
+ * @param programId
+ * @param donorId
+ * @returns
+ */
+export async function getDonor(programId: string, donorId: string): Promise<ClinicalDonor | undefined> {
+  return DONOR_FETCH_CACHE.get({ programId, donorId });
+}
```

Comment:
`getDonor` is now used in place of `fetchDonor`. `fetchDonor` has all the code for calling clinical for donor data, while `getDonor` uses `fetchDonor` and caches the results using the new `AsyncCache` utility.